libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -424,27 +423,15 @@ class MapboxNavigation(
                     }
                 )
 
-                val bearingElements = options.bearings()?.split(";")
-                val originTolerance =
-                    bearingElements?.getOrNull(0)?.split(",")?.getOrNull(1)?.toDouble()
-                bearingElements?.subList(index + 1, coordinates.size)?.map { element ->
-                    element.split(",").let { components ->
-                        if (components.size == 2) {
-                            Pair(components[0].toDouble(), components[1].toDouble())
-                        } else {
-                            null
-                        }
-                    }
-                }?.toMutableList()?.also { pairs ->
-                    pairs.add(
-                        0,
-                        Pair(
-                            location.bearing.toDouble(),
-                            originTolerance ?: DEFAULT_REROUTE_BEARING_TOLERANCE
-                        )
-                    )
-                    optionsBuilder.bearings(*pairs.toTypedArray())
-                }
+                val bearings = mutableListOf<List<Double>>()
+
+                val originTolerance = options.bearingsList()?.get(0)?.get(1) ?: DEFAULT_REROUTE_BEARING_TOLERANCE
+                val originAngle = location.bearing.toDouble()
```

Comment:
NIT this is the current angle, not original angle

Reply:
fixed